### PR TITLE
Masp tests

### LINKED ISSUE #3836: Test masp edge cases
We should try to test some masp edge cases:

- Test a block (or even batch) with one shielded action and one MASP tx in both orders and check that the shielded sync command always reconstruct the correct state
- Test a batch containing two masp txs, one successful and one failing (in both orders) and check that the protocol and the shielded sync command work as expected. Also test with an atomic batch

Also test some malicious masp transactions:

- A non-MASP transaction that carries a MASP section (check both the case as-is and the one where the tx manually triggers the MASP VP)
- A valid masp tx that carries more than one masp section (check that both the protocol and the shielded sync command ignore the unused masp sections)
- A valid shielded action tx that also carries a masp section (same checks as above)

### PR DESCRIPTION
**Pull request by grarco**
## Describe your changes

Partially addresses #3836.
Partially addresses #3828.

Adds more integration test for the MASP including some tests with tx batches.
Adds some protests for the MASP vp.

## Checklist before merging 
- [ ] If this PR has some consensus breaking changes, I added the corresponding `breaking::` labels
    - This will require 2 reviewers to approve the changes
- [ ] If this PR requires changes to the docs or specs, a corresponding PR is opened in the `namada-docs` repo
    - Relevant PR if applies: 
- [ ] If this PR affects services such as `namada-indexer` or `namada-masp-indexer`, a corresponding PR is opened in that repo
    - Relevant PR if applies:

### CODE REVIEW

File: crates/shielded_token/src/vp.rs [hunk 1]
```diff
@@ -978,4 +988,157 @@ mod shielded_token_tests {
...
+                &ctx,
+                &tx.batch_ref_tx(&cmt),
+                &keys_changed,
+                &verifiers
+            )
+            .is_ok());
+        }
+
+        // Changing unknown masp keys is not allowed
+        #[test]
```
Comment (batconjurer) [inline crates/shielded_token/src/vp.rs:1097]: fmt doesn't work well inside the proptest macro. This will have to be beautified manually

File: crates/tests/src/integration/masp.rs [hunk 2]
```diff
@@ -3758,3 +3726,698 @@ fn identical_output_descriptions() -> Result<()> {
...
+                CHRISTEL_KEY,
+                "--output-folder-path",
+                tempdir.path().to_str().unwrap(),
+                "--dump-tx",
+                "--ledger-address",
+                validator_one_rpc,
+            ],
+        )?;
+        node.assert_success();
+        let file_path = tempdir
```
Comment (batconjurer) [inline crates/tests/src/integration/masp.rs:3798]: you love to see it

File: crates/tests/src/integration/masp.rs [hunk 3]
```diff
@@ -3758,3 +3726,698 @@ fn identical_output_descriptions() -> Result<()> {
...
+        txs.push(batched_tx.to_bytes());
+    }
+
+    node.clear_results();
+    node.submit_txs(txs);
+
+    // Check the block result
+    {
+        let codes = node.tx_result_codes.lock().unwrap();
+        // If empty than failed in process proposal
```
Comment (batconjurer) [inline crates/tests/src/integration/masp.rs:3862]:
```suggestion
        // If empty then failed in process proposal
```

File: crates/tests/src/integration/masp.rs [hunk 4]
```diff
@@ -3758,3 +3726,698 @@ fn identical_output_descriptions() -> Result<()> {
...
```
Comment (batconjurer) [inline crates/tests/src/integration/masp.rs:4094]:
```suggestion
        // If empty then failed in process proposal
```

**Review by batconjurer — APPROVED** [review]
